packages/components/badge/src/badge.scss

```diff
@@ -105,7 +105,30 @@
   padding: var(--sl-size-025) var(--sl-size-100);
 
   ::slotted(sl-icon) {
-    --sl-icon-size: var(--sl-size-icon-xs);
+    --sl-icon-size: var(--sl-size-new-icon-xs);
+  }
+}
+
+:host([size='sm'][round]) {
+  background: transparent;
+  color: inherit;
+  font-size: 1em;
+  line-height: unset;
+}
+
+:host([size='sm'][round])::before {
+  background: var(--_background, var(--sl-color-background-accent-grey-subtle));
+  block-size: var(--sl-size-100);
+  border-radius: var(--sl-size-borderRadius-full);
```

Comment:
I don't think it makes sense to use a token here; in other places i just use `50%`.